### PR TITLE
BESU-164: Add information about creating indexes for log queries.

### PR DESCRIPTION
**Pull request by bgravenorst**
Signed-off-by: Byron Gravenorst <byron.gravenorst@consensys.net>

## PR description
Add information about creating indexes for log queries.

## Fixed Issue(s)
BESU-164

### CODE REVIEW

File: docs/Reference/API-Methods.md [hunk 1]
```diff
@@ -88,55 +88,107 @@ This example changes the debug level of all logs to `WARN`.
...
+
+`integer` - Block to start generating indexes
+
+`integer` - Block to stop generating indexes
+
+**Returns**
+
+`result` : *object* - Log bloom index details:
+
+* `integer` : `startBlock` - Starting block for the generated indexes
```
Comment (bgravenorst) [inline docs/Reference/API-Methods.md]: I'm unsure about the returns section, could you please confirm these?

Comment (shemnon) [reply docs/Reference/API-Methods.md]:
```suggestion
* `quantity` : `startBlock` - Starting block of the last requested cache generation
```

File: docs/Reference/API-Methods.md [hunk 2]
```diff
@@ -88,55 +88,107 @@ This example changes the debug level of all logs to `WARN`.
...
+* `integer` : `endBlock` - End block for the generated indexes
```
Comment (shemnon) [inline docs/Reference/API-Methods.md]:
```suggestion
* `quantity` : `endBlock` - Ending block of the last requested cache generation
```

File: docs/Reference/API-Methods.md [hunk 3]
```diff
@@ -88,55 +88,107 @@ This example changes the debug level of all logs to `WARN`.
...
+* `quantity` : `currentBlock` - Hexadecimal integer representing the latest
```
Comment (shemnon) [inline docs/Reference/API-Methods.md]: I think calling out the hexideicmal number is already taken care of by calling it a quantity.
```suggestion
* `quantity` : `currentBlock` - The most recent block that was added to the cache
```

File: docs/Reference/API-Methods.md [hunk 4]
```diff
@@ -88,55 +88,107 @@ This example changes the debug level of all logs to `WARN`.
...
+   block indexed
+* `boolean` : `indexing` - `true` if indexing is in progress
+* `boolean` : `requestAccepted` - `true` if the request to generate log bloom
```
Comment (shemnon) [inline docs/Reference/API-Methods.md]:
```suggestion
* `boolean` : `requestAccepted` - `true` if the request from this call to generate log bloom cache has been accepted
```

File: docs/Reference/CLI/CLI-Subcommands.md [hunk 5]
```diff
@@ -109,10 +109,31 @@ besu operator generate-blockchain-config --config-file=<FILE> --to=<DIRECTORY> [
...
 
-The configuration file has 2 subnested JSON nodes. The first is the `genesis` property defining 
-the [IBFT 2.0 genesis file](../../HowTo/Configure/Consensus-Protocols/IBFT.md#genesis-file) except for the `extraData` string. The 
+The configuration file has 2 subnested JSON nodes. The first is the `genesis` property defining
+the [IBFT 2.0 genesis file](../../HowTo/Configure/Consensus-Protocols/IBFT.md#genesis-file) except for the `extraData` string. The
 second is the `blockchain` property defining the number of key pairs to generate.  
 
+### generate-log-bloom-cache
+
+Generates cached log bloom indexes for blocks. APIs use the cached indexes for improved log
```
Comment (shemnon) [inline docs/Reference/CLI/CLI-Subcommands.md]: Do we want to note that this cannot be done against a running node or is that implicit?

Comment (bgravenorst) [reply docs/Reference/CLI/CLI-Subcommands.md]: @MadelineMurray, do we expect users to know this? For example we don't call this out for `public-key export-address`, but maybe we should?

File: docs/Reference/CLI/CLI-Subcommands.md [hunk 6]
```diff
@@ -109,10 +109,31 @@ besu operator generate-blockchain-config --config-file=<FILE> --to=<DIRECTORY> [
...
+query performance.
+
+!!! note
+    Each index file contains 100000 blocks. The last fragment of blocks less that 100000 are not indexed.
+
+Must be manually executed to add new blocks to the indexes.
+
+The [`admin_generateLogBloomCache`](../API-Methods.md#admin_generateLogBloomCache) API can also be used to generate cached log
```
Comment (MadelineMurray) [inline docs/Reference/CLI/CLI-Subcommands.md]: The deep link (`admin_generateLogBloomCache`) isn't working - needs to be all lower-case `admin_generatelogbloomcache`

File: docs/Concepts/Events-and-Logs.md [hunk 7]
```diff
@@ -2,21 +2,24 @@
...
+be stored in and accessed from logs, the cost is reduced. For example, you can use logs to display all
+transfers made using a specific contract but not the current state of the contract.
 
-A Dapp front end can either request logs using the [JSON-RPC API filter methods](../HowTo/Interact/Filters/Accessing-Logs-Using-JSON-RPC.md) 
-or subscribe to logs using the [RPC Pub/Sub API](../HowTo/Interact/APIs/RPC-PubSub.md#logs). 
+A Dapp front end can either request logs using the [JSON-RPC API filter methods](../HowTo/Interact/Filters/Accessing-Logs-Using-JSON-RPC.md)
+or subscribe to logs using the [RPC Pub/Sub API](../HowTo/Interact/APIs/RPC-PubSub.md#logs).
 
-## Topics 
+Improve log retrieval performance by using [`admin_generateLogBloomCache`](../Reference/API-Methods.md#admin_generatelogbloomcache) to
```
Comment (MadelineMurray) [inline docs/Concepts/Events-and-Logs.md]: Suggest rewording to make active: 

Use [`admin_generateLogBloomCache`](../Reference/API-Methods.md#admin_generatelogbloomcache) to improve log retrieval performance.

File: docs/Reference/API-Methods.md [hunk 8]
```diff
@@ -2590,16 +2642,18 @@ Polls the specified filter and returns an array of changes that have occurred si
...
-    
-    
+
+
 
 ### eth_getFilterLogs
 
 Returns an array of [logs](../Concepts/Events-and-Logs.md) for the specified filter.
 
+Improve log retrieval performance by using [`admin_generateLogBloomCache`](admin_generatelogbloomcache).
```
Comment (MadelineMurray) [inline docs/Reference/API-Methods.md]: Suggest rewording to make active: 

Use [`admin_generateLogBloomCache`](admin_generatelogbloomcache) to improve log retrieval performance.

File: docs/Reference/API-Methods.md [hunk 9]
```diff
@@ -2650,7 +2704,10 @@ Returns an array of [logs](../Concepts/Events-and-Logs.md) for the specified fil
 
 ### eth_getLogs
 
-Returns an array of [logs](../Concepts/Events-and-Logs.md) matching a specified filter object.
+Returns an array of [logs](../Concepts/Events-and-Logs.md) matching a specified
+filter object.
+
+Improve log retrieval performance by using [`admin_generateLogBloomCache`](admin_generatelogbloomcache).
```
Comment (MadelineMurray) [inline docs/Reference/API-Methods.md]: Same comment as above about rewording to make active.

File: docs/Reference/API-Methods.md (hunk 1 above)
Comment (bgravenorst) [reply docs/Reference/API-Methods.md]: Done

File: docs/Reference/CLI/CLI-Subcommands.md (hunk 6 above)
Comment (bgravenorst) [reply docs/Reference/CLI/CLI-Subcommands.md]: Fixed.
